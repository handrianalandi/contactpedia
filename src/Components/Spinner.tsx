import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSpinner } from "@fortawesome/free-solid-svg-icons";
import styled from "@emotion/styled";

interface SpinnerProps {
  isInvertedColor?: boolean;
  size?: "xs" | "sm" | "lg";
}

const SpinnerElement = styled(FontAwesomeIcon)<SpinnerProps>`
  color: ${(props) => (props.isInvertedColor ? "white" : "#06ba63")};
  font-size: ${(props) =>
    props.size === "xs" ? "1rem" : props.size === "sm" ? "1.5rem" : "2rem"};
`;

export default function Spinner({
  isInvertedColor = false,
  size = "xs",
}: SpinnerProps) {
  return (
    <div>
      <SpinnerElement
        size={size}
        isInvertedColor={isInvertedColor}
        icon={faSpinner}
        spin
      />
    </div>
  );
}