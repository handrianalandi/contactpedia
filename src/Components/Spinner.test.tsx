import { render } from "@testing-library/react";
import Spinner from "./Spinner";
describe("Spinner", () => {
  const sizes: ("xs" | "sm" | "lg")[] = ["xs", "sm", "lg"];
  describe("should render correctly", () => {
    sizes.forEach((size) => {
      it(`with size ${size}`, () => {
        const { container } = render(<Spinner size={size} />);
        expect(container).toMatchSnapshot();
      });
    });
  });

  it("should render correctly with isInvertedColor", () => {
    const { container } = render(<Spinner isInvertedColor />);
    expect(container).toMatchSnapshot();
  });
});
