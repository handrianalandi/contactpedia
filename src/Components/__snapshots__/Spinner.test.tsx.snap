// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render correctly with isInvertedColor 1`] = `
.emotion-0 {
  color: white;
  font-size: 1rem;
}

<div>
  <div>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-spinner fa-spin fa-xs  emotion-0"
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`Spinner should render correctly with size lg 1`] = `
.emotion-0 {
  color: #06ba63;
  font-size: 2rem;
}

<div>
  <div>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-spinner fa-spin fa-lg  emotion-0"
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`Spinner should render correctly with size sm 1`] = `
.emotion-0 {
  color: #06ba63;
  font-size: 1.5rem;
}

<div>
  <div>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-spinner fa-spin fa-sm  emotion-0"
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`Spinner should render correctly with size xs 1`] = `
.emotion-0 {
  color: #06ba63;
  font-size: 1rem;
}

<div>
  <div>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-spinner fa-spin fa-xs  emotion-0"
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
